cpp/tests/community/k_truss_test.cu

```diff
@@ -0,0 +1,408 @@
+          }
+
+          std::set<vertex_t> nbr_intersection_u_v;
+          // Find the intersection of nbr_u and nbr_v
+          std::set_intersection(nbrs_u.begin(),
+                                nbrs_u.end(),
+                                nbrs_v.begin(),
+                                nbrs_v.end(),
+                                std::inserter(nbr_intersection_u_v, nbr_intersection_u_v.end()));
+
+          if (nbr_intersection_u_v.size() < (k - 2)) {
+            auto edge = thrust::make_tuple(vertex_t{*u}, vertex_t{*v});
+
+            if (edge_weight_view) {
+              auto weighted_edge_last = thrust::remove_if(
+                thrust::host,
+                weighted_edge_first,
+                weighted_edge_first + h_src.size(),
+                [edge](auto e) {
+                  auto src         = thrust::get<0>(e);
+                  auto dst         = thrust::get<1>(e);
+                  auto remove_edge = thrust::make_tuple(src, dst);
+                  return (remove_edge == edge || thrust::make_tuple(dst, src) == edge);
+                }
+
+              );
+
+              h_src.resize(thrust::distance(weighted_edge_first, weighted_edge_last));
+              h_dst.resize(thrust::distance(weighted_edge_first, weighted_edge_last));
+              h_wgt.resize(thrust::distance(weighted_edge_first, weighted_edge_last));
+            }
+            auto edge_last = thrust::remove_if(
+              thrust::host,
+              edge_first,
+              edge_first + h_src.size(),
+              [edge](auto e) {
+                auto src         = thrust::get<0>(e);
+                auto dst         = thrust::get<1>(e);
+                auto remove_edge = thrust::make_tuple(src, dst);
+                return (remove_edge == edge || thrust::make_tuple(dst, src) == edge);
+              }
+
+            );
+
+            h_src.resize(thrust::distance(edge_first, edge_last));
+            h_dst.resize(thrust::distance(edge_first, edge_last));
+
+            n_dropped += 1;
+          }
+        }
+      }
+    }
+
+    // convert back to device_vector
+    rmm::device_uvector<vertex_t> d_reference_src(h_src.size(), handle.get_stream());
+
+    raft::update_device(d_reference_src.data(), h_src.data(), h_src.size(), handle.get_stream());
+
+    rmm::device_uvector<vertex_t> d_reference_dst(h_src.size(), handle.get_stream());
+
+    raft::update_device(d_reference_dst.data(), h_dst.data(), h_dst.size(), handle.get_stream());
+
+    auto d_reference_wgt =
+      std::make_optional<rmm::device_uvector<weight_t>>(h_src.size(), handle.get_stream());
+
+    if (edge_weight_view) {
+      raft::update_device(
+        (*d_reference_wgt).data(), h_wgt.data(), h_wgt.size(), handle.get_stream());
+    }
+
+    return std::make_tuple(std::move(d_reference_src),
+                           std::move(d_reference_dst),
+                           edge_weight_view ? std::move(d_reference_wgt) : std::nullopt);
```

Comment:
You can do this outside the reference implementation. And cugrpah::test has `to_host` and `to_device` utility functions

Reply:
Right!  I figured it later but now I am no longer making any copy and now retrieving host values through `graph_to_host_csr`.